web/src/main/java/com/epam/esm/controller/CertificateController.java

```diff
@@ -0,0 +1,77 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.dto.CertificateDTO;
+import com.epam.esm.dto.query.CertificatePageQueryDTO;
+import com.epam.esm.service.CertificateService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.validation.Valid;
+import javax.validation.constraints.Size;
+import java.util.List;
+
+@RestController
+@RequestMapping(value = "/certificates")
+@Validated
+public class CertificateController {
+    private final CertificateService certificateService;
+
+    @Autowired
+    public CertificateController(CertificateService certificateService) {
+        this.certificateService = certificateService;
+    }
+
+    @GetMapping()
+    @ResponseStatus(HttpStatus.OK)
+    public List<CertificateDTO> find(@Size(max = 50) @RequestParam(required = false) String tagName,
```

Comment:
Use `CertificatePageQueryDTO queryDTO`  as method paramether and do validation and defaults in DTO class

Reply:
Fixed it